packages/block-editor/src/store/reducer.js

```diff
@@ -699,6 +699,27 @@ export function isCaretWithinFormattedText( state = false, action ) {
 }
 
 const BLOCK_SELECTION_EMPTY_OBJECT = {};
```

Comment:
A few notes on my documentation attempts here (cc @ellatrix):

- I assumed `BLOCK_SELECTION_EMPTY_OBJECT` was used here to allow for some strict equality checks in selectors, though I could not find any place we actually do this. The tests also pass if its use below in `BLOCK_SELECTION_INITIAL_STATE` are replaced with new objects. Can you clarify what purpose this value serves?
- And on that note, for what reason does it need to be an object? I had planned to document `start` and `end` as an object, but could only find that we reference `clientId` on it. Are there other properties, or could it just be replaced with a direct reference to the `clientId`?

Reply:
Don't we do any strict equality checks in the reducer? Perhaps I removed that later on, so this wouldn't be needed any longer.

The selection object can also include the rich text identifier and caret position, if a rich text instance is selected.